### PR TITLE
Add options needed to build plugins

### PR DESCRIPTION
**Pull request by gdesmott**
See #135 for the context and API discussion.

### CODE REVIEW

**Review by lu-zero — APPROVED** [review]
It seems good to me, I'm a little busy today so I'd merge it once I give it a go on sled, rav1e and nnnoiseless. If you already tested it it can be merged now.

**Comment by sdroege** [comment]
Also ebur128 please, it makes use of many of the other configurations that are hopefully not affected by this change :)

**Comment by gdesmott** [comment]
I never build or use any of those crates so I prefer to let you double check I didn't break anything. :)

**Comment by gdesmott** [comment]
Just found out that `Libs` field in the pkg-config file should be updated with the subdirectory as well. Let me fix that before merging. :)

**Comment by gdesmott** [comment]
> Just found out that `Libs` field in the pkg-config file should be updated with the subdirectory as well. Let me fix that before merging. :)

Fixed.

I also stopped settings `Cflags` when we are not generating any headers and added a extra commit preventing the creation of `bin` directory if not needed.

File: src/pkg_config_gen.rs [hunk 1]
```diff
@@ -51,6 +51,23 @@ impl PkgConfig {
             _ => Vec::new(),
         };
 
+        let libdir = if let Some(subdir) = &capi_config.library.install_subdir {
+            format!("{}{}{}", "${libdir}", MAIN_SEPARATOR, subdir)
```
Comment (lu-zero) [inline src/pkg_config_gen.rs:55]: probably we could avoid using `format!` and just append to the path and convert later.

Comment (gdesmott) [reply src/pkg_config_gen.rs]: Indeed that's easier, done.

**Pull request merged**